tests/vector_stores/test_pinecone.py

```diff
@@ -0,0 +1,88 @@
+import builtins
+import unittest
+from typing import Any, Callable, Type
+from unittest.mock import patch
+
+import pytest
+from llama_index.vector_stores.pinecone import (
+    PineconeVectorStore,
+)
+
+
+class MockPineconePods:
+    __version__ = "2.2.4"
+
+    @staticmethod
+    def init(api_key: str, environment: str) -> None:
+        pass
+
+    class Index:
+        def __init__(self, index_name: str) -> None:
+            pass
+
+
+class MockPineconeServerless:
+    __version__ = "3.0.0"
+
+    class Pinecone:
+        def __init__(self, api_key: str) -> None:
+            pass
+
+        class Index:
+            def __init__(self, index_name: str) -> None:
+                pass
+
+
+class MockUnVersionedPineconeRelease:
+    @staticmethod
+    def init(api_key: str, environment: str) -> None:
+        pass
+
+    class Index:
+        def __init__(self, index_name: str) -> None:
+            pass
+
+
+def get_version_attr_from_mock_classes(mock_class: Type[Any]) -> None:
+    if not hasattr(mock_class, "__version__"):
+        raise AttributeError(
+            "The version of pinecone you are using does not contain necessary __version__ attribute."
+        )
+
+
+def mock_import(name: str, *args: Any, **kwargs: Any) -> Callable:
+    if name == "pinecone":
+        return MockPineconePods if pods_version else MockPineconeServerless  # type: ignore[name-defined]
+    return original_import(name, *args, **kwargs)  # type: ignore[name-defined]
+
+
+class TestPineconeVectorStore(unittest.TestCase):
+    def setUp(self) -> None:
+        global original_import
+        original_import = builtins.__import__  # type: ignore[name-defined]
+
+    def tearDown(self) -> None:
+        builtins.__import__ = original_import  # type: ignore[name-defined]
+
+    def test_pods_version(self) -> None:
+        global pods_version
+        pods_version = True  # type: ignore[name-defined]
+        with patch("builtins.__import__", side_effect=mock_import):
+            get_version_attr_from_mock_classes(MockPineconePods)
+            store = PineconeVectorStore(
+                api_key="dummy_key", index_name="dummy_index", environment="dummy_env"
+            )
+
+    def test_serverless_version(self) -> None:
+        global pods_version
+        pods_version = False  # type: ignore[name-defined]
+        with patch("builtins.__import__", side_effect=mock_import):
+            get_version_attr_from_mock_classes(MockPineconeServerless)
+            store = PineconeVectorStore(api_key="dummy_key", index_name="dummy_index")
+
+    def test_unversioned_pinecone_client(self) -> None:
```

Comment:
@zackproser test to ensure pinecone client has `__version__` attribute from now on :)

Reply:
(I think technically this type of test should be on the client side -- and Jen says it is now -- but putting here just in case. LMK if you think that's overkill / could actually do more harm than good in the future)

Reply:
Sounds good!